lib/src/rules/iterable_contains_unrelated_type.dart

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.iterable_contains_unrelated_type;
+
+import 'package:analyzer/analyzer.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:linter/src/linter.dart';
+import 'package:linter/src/rules/unrelated_type_equality_checks.dart';
+
+const _desc = r'Invocation of Iterable<E>.contains with references of unrelated'
+    r' types.';
+
+const _details = r'''
+
+**DON'T** Invoke contains on Iterable with an instance of different type than
+the parameter type since it will invoke == on its elements and most likely will
```

Comment:
And here:

the parameter type since it will invoke `==` on its elements and most likely will